src/compiler.rkt

```diff
@@ -48,30 +123,60 @@
     (define size 0)
     (define exprc 0)
     (define varc (length vars))
-
+    
     ; Translates programs into an instruction sequence
-    (define (munge prog type)
+    (define (munge-ival prog type [prec 0])
+      (set! size (+ 1 size))
+      
+      (define expr
+        (match prog
+          [(? number?) (list (const (input->value prog type)))]
+          [(? variable?) prog]
+          [`(if ,c ,t ,f)
+           (list (vector if-proc prec)
+                 (munge-ival c cond-type prec)
+                 (munge-ival t type prec)
+                 (munge-ival f type prec))]
+          [(list op args ...)
+           #:when (set-member? '(sin cos tan) op)
```

Comment:
Better to write:

```suggestion
          [(list (and (or 'sin 'cos 'tan) op) args ...)
```

Reply:
And I think if we add a `#:when (equal? name 'ival)` here I think we don't need two `munge` functions.

Reply:
Agree, I am just afraid that if we will tune more operations in future than `munge` for floats will go through `#:when (equal? name 'ival)` many times

Reply:
I think the speed of `munge` doesn't matter much.